recipes/openssh/all/conanfile.py

```diff
@@ -0,0 +1,130 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import cross_building
+from conan.tools.env import Environment, VirtualBuildEnv, VirtualRunEnv
+from conan.tools.files import patch, copy, get, rm, rmdir
+from conan.tools.gnu import Autotools, AutotoolsToolchain, AutotoolsDeps
+from conan.tools.layout import basic_layout
+from os.path import join
+
+
+required_conan_version = ">=1.54.0"
+
+class PackageConan(ConanFile):
+    name = "openssh"
+    description = "The OpenSSH (portable) suite of secure connectivity tools"
+    license = "BSD"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/openssh/openssh-portable"
+    topics = ("security", "cryptography", "login", "keychain", "file-sharing")
+    package_type = "application"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "with_openssl": [True, False],
+        "with_pam": [True, False],
+        "with_sandbox": ["auto", "no", "capsicum", "darwin", "rlimit", "seccomp_filter", "systrace", "pledge"]
+    }
+    default_options = {
+        "with_openssl": True,
+        "with_pam": False,
+        "with_sandbox": "auto"
+    }
+
+    def _patch_sources(self):
+        print("patching for {}-{}-{}".format(self.version, str(self.settings.os).lower(), self.settings.get_safe("os.version")))
+
+        # general patches
+        for p in self.conan_data.get("patches", {}).get(self.version, []):
+            print(f"patching general files: {p['patch_file']}")
+            patch(self, **p, base_path=self.source_folder)
+        
+        # os specific patches
+        for p in self.conan_data.get("patches", {}).get("{}-{}".format(self.version, str(self.settings.os).lower()), []):
+            print(f"patching os specific files: {p['patch_file']}")
+            patch(self, **p, base_path=self.source_folder)
+
+        # os version specific patches
+        for p in self.conan_data.get("patches", {}).get("{}-{}-{}".format(self.version, str(self.settings.os).lower(), self.settings.get_safe("os.version")), []):
+            print(f"patching os version specific: {p['patch_file']}")
+            patch(self, **p, base_path=self.source_folder)
+
+    def package_id(self):
+        del self.info.settings.compiler
+        del self.info.settings.build_type
+
+    def export_sources(self):
+        copy(self, "patches/*.patch", self.recipe_folder, self.export_sources_folder, keep_path=True)
+
+    def configure(self):
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("zlib/[>=1.2.12]")
```

Comment:
```suggestion
        self.requires("zlib/[>=1.2.13 <2]")
```

We're planning on adding zlib to the list of allowed recipes with evrsion ranges as we being to feel more confident that version ranges are the right aapproach, but we're not quite yet there.

BUT, as this might still take a few days to get applied, I'm suggesting the provisional range for zlib :)